core/src/main/scala/scalaz/FreeT.scala

```diff
@@ -23,7 +23,11 @@ object FreeT extends FreeTInstances {
   /** Return the given value in the free monad. */
   def point[S[_], M[_], A](value: A)(implicit M: Applicative[M]): FreeT[S, M, A] = Suspend(M.point(-\/(value)))
 
-  def suspend[S[_], M[_], A](a: M[A \/ S[FreeT[S, M, A]]]): FreeT[S, M, A] = Suspend(a)
+  def suspend[S[_], M[_], A](a: M[A \/ S[FreeT[S, M, A]]])(implicit M: Applicative[M]): FreeT[S, M, A] =
+    liftM(a).flatMap({
+      case -\/(a) => point(a)
+      case \/-(s) => roll(s)
+    })
```

Comment:
`suspend` with this signature doesn't seem like a very useful smart constructor for the user. Perhaps it was never meant to be public? It used to mimic (leak?) the representation of `Suspend`, which this commit changes anyway. Should `suspend`, with this signature, be removed?